File: benchmark/list_insert.py
import time

n = 35000
t = time.clock()
sum = 0

a, b, c = [], [], []
for i in range(n):
	a.append(i)
	b.insert(0, i)
	c.insert(i//2, i)
	sum += a[0] + b[0] + c[0] + a[-1] + b[-1] + c[-1]


t = time.clock()-t
print('Sum: '+str(sum))
print('Elapsed: '+str(t))

File: benchmark/list_insert.lua
local n = 35000
local t = os.clock()
local sum = 0
local insert, ceil = table.insert, math.ceil

local a, b, c = {}, {}, {}
for i = 1, n  do
insert(a, i)
insert(b, 1, i)
insert(c, ceil(i/2), i)
sum = sum + a[1] + b[1] + c[1] + a[i] + b[i] + c[i]
end

t = os.clock()-t
io.write('Sum: ' ..sum .. '\n')
io.write(string.format("elapsed: %.8f\n", t))